##### Combining DataFrames and Missing Values
###### As part of its forthcoming holiday campaign, the business wanted to understand the cost of acquisition of customers for an e-commerce website specializing in kids' toys.
###### As a marketing analyst, you now have to dig through the historical records of the previous campaign and suggest the market budget for the current campaign.

In [1]:
# Import pandas modules

import pandas as pd

In [2]:
# Load the CSV files

df1 = pd.read_csv("timeSpent.csv")

df2 = pd.read_csv("cost.csv")

In [3]:
# Examine the first few rows for timespent data
df1.head()

users   timeSpent
0  Sushmita   31.266056
1      Adam   17.900529
2     Benny  265.236086
3    Anurag   42.470696

In [4]:
# Examine the first few rows for cost data
df2.head()

users  cost
0      Adam  20.0
1    Anurag   NaN
2     Benny  15.0
3  Sushmita   2.0
4   Apoorva   7.0

In [6]:
# Perform a left join of df1 and df2
df = df1.merge(df2, on="users", how="left")
df.head()

users   timeSpent  cost
0  Sushmita   31.266056   2.0
1      Adam   17.900529  20.0
2     Benny  265.236086  15.0
3    Anurag   42.470696   NaN

In [7]:
# Replace the missing value with 0 
df = df.fillna(0)
df.head()

users   timeSpent  cost
0  Sushmita   31.266056   2.0
1      Adam   17.900529  20.0
2     Benny  265.236086  15.0
3    Anurag   42.470696   0.0

In [8]:
# Compute the average cost of acquisition along with the everage time spent.
df.describe()

timeSpent       cost
count    4.000000   4.000000
mean    89.218342   9.250000
std    117.774200   9.776673
min     17.900529   0.000000
25%     27.924674   1.500000
50%     36.868376   8.500000
75%     98.162044  16.250000
max    265.236086  20.000000

###### The everage cost of acquisition of a user is $9.25.
###### Also, time spent by a user is 89 seconds on the website. 